### PR TITLE
DOCSP-31190 Zero value for missing required property

### PR DESCRIPTION
**Pull request by lindseymoore**
### Jira

- https://jira.mongodb.org/browse/DOCSP-31190

### Staged Changes

- [Note in Required Property section](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-31190/sync/data-model/update-schema/#add-a-required-property)
- [Addition to Default Row in Property Types table](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-31190/sync/data-model/data-model-map/#property-types)

### Reminder Checklist

You might need to also update some corresponding pages. Check if completed or N/A.

- [ ] Create Jira ticket for corresponding docs-realm update(s), if any
- [ ] Checked/updated Admin API
- [ ] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sync/data-model/update-schema.txt
Comment (dacharyc) [inline source/sync/data-model/update-schema.txt:181]: Nit: our style guide calls for us to avoid future tense.

```suggestion
.. note:: Objects missing a required property default to a zero value
```

File: source/sync/data-model/update-schema.txt
Comment (dacharyc) [inline source/sync/data-model/update-schema.txt:184]: Nit: tense.
```suggestion
   The server populates the missing required value field with a zero or blank value
```

**Review by dacharyc — COMMENTED** [review]
Two general comments:

- This page is about changing a data model. I think this behavior is general to _any_ required property - not just adding one when you update the data model - so it might be good to also include this note on the "[Data Model Mapping](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-31190/sync/data-model/data-model-map/#property-types)" page.
- The "Property Types" table on the Data Model Mapping page has a "Default" parameter that explicitly says "When you attempt to create an object that is missing a value for a required field, it fails validation and does not persist to the realm." This note seems to contradict that. It would be good to confirm which behavior is expected, and update the verbiage on the page if needed.

File: source/sync/data-model/update-schema.txt
Comment (dacharyc) [inline source/sync/data-model/update-schema.txt:187]: I can't comment directly, but the section below this - "Modify Existing Atlas Documents" - describes how you have to update existing documents with a new required property or they don't sync with the client. 

Is that still true with this new required property behavior? i.e. does the server populate missing required value fields only for _new_ documents, or does it also retroactively apply the missing default for documents that are already in a collection?

If it's true only for _new_ documents, we should probably add that in the note. Maybe something like "When you sync objects from the client that are missing a required property, the server populates the missing required value field with..."

And then maybe add clarification in the section below, too, that this behavior doesn't apply to existing documents in a collection, and they still have to populate those with default values or they don't sync.

OR if they no longer have to add default values to existing documents to get them to sync, we should probably delete the section below?

**Review by dacharyc — COMMENTED** [review]
One more question about something that we should probably clarify before we merge.

File: source/sync/data-model/update-schema.txt
Comment (lindseymoore) [reply source/sync/data-model/update-schema.txt:187]: It's only for new documents! I think specifying in the note will suffice. Wouldn't want to convolute the next section for such a small case. Let me know what you think of the update!

Comment (dacharyc) [reply source/sync/data-model/update-schema.txt:187]: Perfect, that makes sense. Thank you for clarifying!

**Review by dacharyc — APPROVED** [review]
LGTM!

File: source/sync/data-model/update-schema.txt
Comment (benredmond) [inline source/sync/data-model/update-schema.txt:181]: Sorry I think I misunderstood your question on slack about this - we actually do this every time for data coming from Realm. this section LGTM if we just remove the references to "new"

**Review by benredmond — APPROVED** [review]
LGTM aside from a clarification on the "new" documents coming from Realm. Thanks for picking this up! I'm happy that we're making this behavior more explicit to the user :)

**Pull request merged**
